src/phoenix/trace/llama_index/callback.py

```diff
@@ -18,21 +17,43 @@
 
 
 def _check_instrumentation_compatibility() -> bool:
-    if find_spec("llama_index") is None:
-        raise PackageNotFoundError("Missing `llama-index`. Install with `pip install llama-index`.")
-    # split the version string into a tuple of integers
-    llama_index_version_str = version("llama-index")
-    llama_index_version = tuple(map(int, llama_index_version_str.split(".")[:3]))
+    try:
+        llama_index_version_str = version("llama-index")
+    except PackageNotFoundError:
+        llama_index_version_str = None
+
+    try:
+        llama_index_core_version_str = version("llama-index-core")
+    except PackageNotFoundError:
+        llama_index_core_version_str = None
+
     instrumentation_version_str = version("openinference-instrumentation-llama-index")
-    instrumentation_version = tuple(map(int, instrumentation_version_str.split(".")[:3]))
-    # check if the llama_index version is compatible with the instrumentation version
+    instrumentation_version = _parse_semantic_version(instrumentation_version_str)
+
+    if llama_index_version_str is None:
+        if llama_index_core_version_str is None:
+            raise PackageNotFoundError(
+                "Missing `llama_index`. "
+                "Install with `pip install llama-index` or "
+                "`pip install llama-index-core` for a minimal installation."
+            )
```

Comment:
why not just combine these conditionals with an or?

Reply:
I could combine with an `and`, but then I would need to do an additional conditional to handle the case where `llama-index-core` is installed but the instrumentation is out of date.

There are four cases I was checking for:

- The user has neither `llama-index` nor `llama-index-core` installed.
- The user has only `llama-index-core` installed, but an old version of the instrumentation.
- The user has an old version of `llama-index` installed and a modern version of the instrumentation.
- The user has a modern version of `llama-index` installed and an old version of the instrumentation.